### PR TITLE
[deconz] Added Water Leakage Sensor, Added checks for optional Channels

### LINKED ISSUE #4515: [deconz] Capture temperature provided by open/close sensors
Within the config section, Xiaomi open/close sensors provide the temperature. It would be great to capture this as well. Note that this cannot be captured with the regular temperature sensors things existing already.

```
"18": {
    "config":    ###{
        "battery": 100,
        "on": true,
        "reachable": true,
        "temperature": 0
    },
    "ep": 1,
    "etag": "d4e8e099a565f6326b9a87b3615ee483",
    "manufacturername": "LUMI",
    "modelid": "lumi.sensor_magnet",
    "name": "OpenClose",
    "state": {
        "lastupdated": "2018-12-16T00:02:47",
        "open": false
    },
    "type": "ZHAOpenClose",
    "uniqueid": "00:15:8d:00:01:f3:87:a2-01-0006"
}
``` 

In addition, capturing the battery level would be the icing on the cake. Thanks. 

Related to #4400

### LINKED ISSUE #4588: [deconz] Provide support for water leakage sensors
It would be super also to support this one for water leakage detection. DECONZ provides this already. Thanks. 

`{"config":"battery":100,"on":true,"reachable":true,"temperature":3000},"ep":1,"etag":"xxx","manufacturername":"LUMI","modelid":"lumi.sensor_wleak.aq1","name":"Water",
"state":{"lastupdated":"2019-01-13T18:38:36","lowbattery":false,"tampered":false,"water":false},"type":"ZHAWater","uniqueid":"xxxx"}`

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added support for Water Leakage Sensor
- Added support for optional channel temperature for Xiaomi sensors
- Added support for optional channel power on consumption sensors
- Added support for optional channels voltage and current on power sensors

Closes #4588
Closes #4515

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

**Comment by SuperSchelli** [comment]
Oh, great, i was thinking about fixing those issues. 

Although i am not a reviewer, looks good to me and thx for fixing the typo.

**Comment by cweitkamp** [comment]
@wborn @kaikreuzer The review requests worked for deCONZ binding. Even after a new commit.

**Comment by kaikreuzer** [comment]
Nice 😄

File: addons/binding/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/SensorThingHandler.java [hunk 1]
```diff
@@ -188,10 +182,43 @@ public void bridgeStatusChanged(@NonNull ThingStatusInfo bridgeStatusInfo) {
...
+                                new QuantityType<Temperature>(temperature / 100, SIUnits.CELSIUS));
+                    }
+
+                    // ZHAPresence - e.g. IKEA TRÅDFRI motion sensor
                     if (newState.state.dark instanceof Boolean) {
                         createChannel(CHANNEL_DARK);
                     }
 
+                    // ZHAConsumption - e.g Bitron 902010/25 or Heiman SmartPlug
+                    if (newState.state.power instanceof Float) {
```
Comment (davidgraeff) [inline addons/binding/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/SensorThingHandler.java]: Why is that `instanceof` and not `!= null`, am I missing something?

Comment (cweitkamp) [reply addons/binding/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/SensorThingHandler.java]: Nothing specific. I first thought this is the better condition, but I does not matter. I can change it to `!= null` if you prefer.

Comment (davidgraeff) [reply addons/binding/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/SensorThingHandler.java]: In early java editions `instanceof` had a considerable runtime cost. I guess that's not the case anymore. But why making the runtime more effort if not necessary. And consistency is an argument, as `!= null` is used in other cases.

Comment (cweitkamp) [reply addons/binding/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/SensorThingHandler.java]: Agreed. Done.

**Pull request merged**